presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -465,43 +482,62 @@ protected Scope visitCreateView(CreateView node, Optional<Scope> scope)
         }
 
         @Override
-        protected Scope visitSetSession(SetSession node, Optional<Scope> scope)
+        protected Scope visitTableProperties(TableProperties node, Optional<Scope> scope)
         {
+            Optional<String> duplicateProperty = node.getProperties().stream()
+                    .map(TableProperty::getName)
+                    .map(Identifier::getValue)
+                    .collect(groupingBy(identity(), counting()))
+                    .entrySet()
+                    .stream()
+                    .filter(entry -> entry.getValue() > 1)
+                    .map(Entry::getKey)
+                    .findAny();
+            if (duplicateProperty.isPresent()) {
+                throw new SemanticException(DUPLICATE_TABLE_PROPERTIES, node, "Duplicate table property '%s'", duplicateProperty.get());
```

Comment:
The origin implementation creates a property-name-to-expression map during parsing, and duplication is de-duped silently, which is incorrectly. Separating the change will require rewrite the incorrect old behavior. I don't think that's desirable?